client/state/sites/domains/reducer.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SITE_DOMAINS_RECEIVE,
+	SITE_DOMAINS_REQUEST,
+	SITE_DOMAINS_REQUEST_SUCCESS,
+	SITE_DOMAINS_REQUEST_FAILURE,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+/**
+ * Domains `Reducer` function
+ *
+ * @param {Object} state - current state
+ * @param {Object} action - domains action
+ * @return {Object} updated state
+ */
+export const items = ( state = {}, action ) => {
+	const { siteId } = action;
+	switch ( action.type ) {
+		case SITE_DOMAINS_RECEIVE:
+			return Object.assign(
+				{},
+				state,
+				{
+					[ siteId ]: action.domains
+				}
+			);
```

Comment:
Since we don't have reducer middleware yet, if we leave out the SERIALIZE or DESERIALIZE actions we'll persist by default. To protect against [data drift](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#problem-data-shapes-change-over-time--3101-) we'll want to add a section like:
```javascript
        case DESERIALIZE:
            if ( isValidStateWithSchema( state, itemsSchema ) ) {
                return state;
            }
            return {};
```
Ping me if you need help with the json schema.